cmd/process-agent/main_common.go

```diff
@@ -50,6 +50,36 @@ var (
 	GoVersion string
 )
 
+var rootCmd = &cobra.Command{
+	Run: func(cmd *cobra.Command, args []string) {
+		exit := make(chan struct{})
+
+		// Invoke the Agent
+		runAgent(exit)
+	},
+}
+
+// fixDeprecatedFlags modifies os.Args so that non-posix flags are converted to posix flags
+// it also displays a warning when a non-posix flag is found
+func fixDeprecatedFlags() {
+	deprecatedFlags := map[string]struct{}{
+		// Global flags
+		"-config": {}, "-ddconfig": {}, "-sysprobe-config": {}, "-pid": {}, "-info": {}, "-version": {}, "-check": {},
+
+		// Windows flags
+		"-install-service": {}, "-uninstall-service": {}, "-start-service": {}, "-stop-service": {}, "-foreground": {},
+	}
+
+	for i, a := range os.Args {
+		if _, ok := deprecatedFlags[a]; !ok {
+			continue
+		}
+		fmt.Printf("WARNING: `%s` argument is deprecated and will be removed in a future version. "+
```

Comment:
nit: let's not use `+` for the new line, you can put everything in one line.

Reply:
Fixed!